core/services/network_checker.rb

```diff
@@ -23,11 +23,14 @@ def self.resources_available?(machine_configurator, machine, logger)
            else
              :wget
            end
-    result = resources.all? do |resource|
-      check_resource?(tool, machine_configurator, machine, resource, logger)
+    table = {}
+    resources.each do |resource|
```

Comment:
There is a special iterators. You can either use `reduce` or `each_with_object` to achieve the same result. I suggest the latter one.